num2words/lang_RU.py

```diff
@@ -208,21 +217,43 @@ def to_ordinal(self, number):
                 lastword = lastword[:lastword.rfind('н') + 1] + "ный"
             elif lastword[-1] == "д" or lastword[-2] == "д":
                 lastword = lastword[:lastword.rfind('д') + 1] + "ный"
+
+        if gender == 'f':
+            if lastword[-2:] == "ий":
+                lastword = lastword[:-2] + "ья"
+            else:
+                lastword = lastword[:-2] + "ая"
+        if gender == 'n':
+            if lastword[-2:] == "ий":
+                lastword = lastword[:-2] + "ье"
+            else:
+                lastword = lastword[:-2] + "ое"
+
         outwords[-1] = self.title(lastword)
         return " ".join(outwords).strip()
 
     def _money_verbose(self, number, currency):
-        return self._int2word(number, currency == 'UAH')
+        if currency == 'UAH':
+            gender = 'f'
+        else:
+            gender = 'm'
+
+        return self._int2word(number, gender)
 
     def _cents_verbose(self, number, currency):
-        return self._int2word(number, currency in ('UAH', 'RUB', 'BYN'))
+        if currency in ('UAH', 'RUB', 'BYN'):
+            gender = 'f'
+        else:
+            gender = 'm'
+
+        return self._int2word(number, gender)
 
-    def _int2word(self, n, feminine=False):
+    def _int2word(self, n, gender):
```

Comment:
for backwards comptability, probably, it make sense to add check like

if isinstance(gender, bool) and gender:
    gender = 'f'

Reply:
Fixed, please check